core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -39,22 +37,22 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
     }
   }
 
-  def starting(work: Option[(Work, ActorRef)], askWorkImmediately: Boolean): Receive = whileWaiting(Starting) orElse {
+  def waitingForRoutee(work: Option[(Work, ActorRef)]): Receive = whileWaiting(Starting) orElse {
```

Comment:
It will reject it, yes, so I think we are ok.  I also think another way to fix this, is to adjust Queue to use a LinkedHashSet so that it cannot contain duplicate refs to a Worker.  LinkedHashSet maintains insertion order, so we might want to look into that as well.